src/test/java/com/easypost/CarbonOffsetTest.java

```diff
@@ -0,0 +1,133 @@
+package com.easypost;
+
+import com.easypost.exception.EasyPostException;
+import com.easypost.model.Fee;
+import com.easypost.model.Rate;
+import com.easypost.model.Shipment;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertInstanceOf;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public final class CarbonOffsetTest {
+    private static TestUtils.VCR vcr;
+
+    /**
+     * Set up the testing environment for this file.
+     *
+     * @throws EasyPostException when the request fails.
+     */
+    @BeforeAll
+    public static void setup() throws EasyPostException {
+        vcr = new TestUtils.VCR("carbon_offset", TestUtils.ApiKey.TEST);
+    }
+
+    /**
+     * Test creating a shipment with a carbon offset.
+     *
+     * @throws EasyPostException when the request fails.
+     */
+    @Test
+    public void testCreateShipmentWithCarbonOffset() throws EasyPostException {
+        vcr.setUpTest("create_shipment_with_carbon_offset");
+
+        Shipment shipment = Shipment.create(Fixture.basicCarbonOffsetShipment(), true);
+
+        assertInstanceOf(Shipment.class, shipment);
+
+        List<Rate> rates = shipment.getRates();
+        assertNotNull(rates);
+
+        Rate rate = rates.get(0);
+        assertNotNull(rate.getCarbonOffset());
```

Comment:
Suggestion: This assertion has no use and should be removed since the one below it assumes its set meaning by only including the assertion on line 48, you get this one for free without the need to include it.